jenkinsapi_utils/jenkins_launcher.py

```diff
@@ -64,9 +74,16 @@ class JenkinsLancher(object):
     JENKINS_WAR_URL = "http://mirrors.jenkins-ci.org/war/latest/jenkins.war"
 
     def __init__(self, war_path, plugin_urls=None, jenkins_url=None):
-        self.jenkins_url = jenkins_url
-        self.http_port = random.randint(9000, 10000) if not jenkins_url \
-            else urlparse(jenkins_url).port
+        if jenkins_url is not None:
+            self.jenkins_url = jenkins_url
+            self.http_port = urlparse(jenkins_url).port
+            self.start_new_instance = False
+        else:
+            self.http_port = random.randint(9000, 10000)
+            self.jenkins_url = 'http://localhost:%s' % self.http_port
```

Comment:
try `sock.bind(('', 0))` and then `sock.getsockname()[1]` to get random port selected by os that guaranteed to be not in use